### PR TITLE
Rails 5 upgrade

### PR DESCRIPTION
**Pull request by Br3nda**
Upgrading Rails -- so then we can use ReactJS in rails.
Then i can start pulling in the WhareHauora-App work by Manu 
Then we'll have only one server (halving the heroku bill)
Then i can enable alert subscription (it's in another PR)
then we'll finish off the new look UI

(we = mostly me, but could be someone we hire too)

## Description
Upgrade to rails 5
needed for #502

### CODE REVIEW

**Comment by Br3nda** [comment]
Had to add some more tests, so our coverage doesn't drop

File: app/models/sensor.rb [hunk 1]
```diff
@@ -30,6 +30,6 @@ def same_home_as_room
   private
 
   def create_room
-    self.room = Room.create(name: mac_address, home: home) if room_id.blank?
+    self.room = Room.create!(name: mac_address, home: home) if room_id.blank?
```
Comment (raquelxmoss) [inline app/models/sensor.rb]: Any extra error handling required here due to the change to a bang method?

File: spec/controllers/admin/home_types_controller_spec.rb [hunk 2]
```diff
@@ -100,29 +100,29 @@
     end
 
     describe 'GET show' do
-      before { get :show, id: home_type.to_param }
+      before { get :show, id: home_type.id }
```
Comment (raquelxmoss) [inline spec/controllers/admin/home_types_controller_spec.rb]: Is this one missing the `params` block? Or does it just happily accept `id` in this form? I assume so, given that the tests pass

Comment (Br3nda) [reply spec/controllers/admin/home_types_controller_spec.rb]: the test were just checking http response code, so i wrote from better ones

**Pull request merged**